SwiftSyntaxDevUtils/Sources/swift-syntax-dev-utils/common/Paths.swift

```diff
@@ -0,0 +1,94 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+enum Paths {
+  static var packageDir: URL {
+    URL(fileURLWithPath: #file)
+      .deletingLastPathComponent()
+      .deletingLastPathComponent()
+      .deletingLastPathComponent()
+      .deletingLastPathComponent()
+      .deletingLastPathComponent()
+  }
+
+  static var sourcesDir: URL {
+    packageDir
+      .appendingPathComponent("Sources")
+  }
+
+  static var examplesDir: URL {
+    packageDir
+      .appendingPathComponent("Examples")
+  }
+
+  static var codeGenerationDir: URL {
+    packageDir
+      .appendingPathComponent("CodeGeneration")
+  }
+
+  static var workspaceDir: URL {
+    packageDir
+      .deletingLastPathComponent()
+  }
+
+  static var llvmDir: URL {
+    workspaceDir
+      .appendingPathComponent("llvm-project")
+      .appendingPathComponent("llvm")
+  }
+
+  static var litExec: URL {
+    llvmDir
+      .appendingPathComponent("utils")
+      .appendingPathComponent("lit")
+      .appendingPathComponent("lit.py")
+  }
+
+  static var python3Exec: URL? {
+    return
+      envSearchPaths
+      .first(where: { $0.path.contains("python3") })
+  }
+
+  static var diffExec: URL? {
+    return
+      envSearchPaths
+      .first(where: { $0.path.contains("diff") })
+  }
```

Comment:
`envSearchPaths` will be the contents of `echo $PATH` in a shell, which is the directory containing executables. Thus, the path itself doesn’t include `diff`. You need to implement something equivalent to `lookupExecutableInPath`

https://github.com/apple/swift-tools-support-core/blob/ef2db24dbe7e39ebfabca2c10905909708c5738b/Sources/TSCBasic/misc.swift#L217-L246